### PR TITLE
feat(react-core[Checkbox]): add pf4 checkbox component

### PR DESCRIPTION
**Pull request by amarie401**
**Affects**: @patternfly/react-core, @patternfly/react-docs

**What**: Introducing the PF4 Check Component

ISSUES CLOSED:  https://github.com/patternfly/patternfly-react/issues/485

### CODE REVIEW

**Comment by priley86** [comment]
hey @amarie401 - just wanted to note that the `/` in the commit subject is still giving our commitizen tool problems 😢 

I recommend just rebasing the commit to something like our others (use a space, hyphen, or bracket instead). Just figured I would share this before the PR is merged 😉

**Comment by amarie401** [comment]
@priley86 thanks for letting me know, will make sure to rebase the commit 👍

File: packages/react-core/src/components/Check/Check.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+const defaultProps = {
+  children: null,
+  className: '',
+  isValid: true,
+  isDisabled: false,
+  onChange: () => undefined
+};
+
+class Check extends React.Component {
```
[the remark this answers is not in the document]
Comment (amarie401) [reply packages/react-core/src/components/Check/Check.js]: I was following the naming convention used in core but can change if we are okay with using Checkbox

**Comment by amarie401** [comment]
Should be good to go now for another review @dmiller9911 @tlabaj

File: packages/react-docs/src/pages/components/checkbox.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    const { data } = this.props;
+    const { checked } = this.state;
+
+    return (
+      <ComponentDocs data={data}>
+        <Example title="Controlled checkbox example">
+          <Checkbox
+            checked={checked}
+            onChange={this.handleCheckboxChange}
+            aria-label="selected checkbox example"
```
Comment (dmiller9911) [inline packages/react-docs/src/pages/components/checkbox.js]: We should probably update these labels too.` "controlled checkbox example"`

File: packages/react-docs/src/pages/components/checkbox.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+          />
+        </Example>
+        <Example title="Uncontrolled checkbox example">
+          <Checkbox aria-label="unselected checkbox example" />
```
Comment (dmiller9911) [inline packages/react-docs/src/pages/components/checkbox.js]: We should probably update these labels too.` "uncontrolled checkbox example"`

**Comment by amarie401** [comment]
@dmiller9911 updated the labels :)

File: packages/react-docs/src/pages/components/checkbox.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    const { data } = this.props;
+    const { checked } = this.state;
+
+    return (
+      <ComponentDocs data={data}>
+        <Example title="Controlled checkbox example">
+          <Checkbox
+            checked={checked}
+            onChange={this.handleCheckboxChange}
+            aria-label="controlled checkbox example"
```
Comment (tlabaj) [inline packages/react-docs/src/pages/components/checkbox.js]: I think you should add the a prop to the Checkbox component for the aria label.
@jgiardino  Should it be required?

Comment (dmiller9911) [reply packages/react-docs/src/pages/components/checkbox.js]: aria-label is good.  Other components will move to it.

**Pull request merged**